### ISSUE TITLE
Retry with byte ranges

### ISSUE THREAD

**Issue by bbockelm** [htcondor/osdf-client#43]
XCache supports byte ranges.  When we fall back to the next source on the list after a failure, instead of starting from offset 0, we should restart at the last good offset.

**Comment by djw8605** [comment]
osdf uses the [grab](https://github.com/cavaliergopher/grab) library which should support download resume by default . Do you have evidence that it isn’t working?  I suppose I never tested it.

**Comment by bbockelm** [comment]
I'll test that!

What made me think it wasn't working was a set of error messages where it appeared the first and second transfer made really significant progress, enough that it should have completed.  However, due to #42, it wasn't quite obvious whether this was true.

**Comment by bbockelm** [comment]
@djw8605 - did some testing.  It does not, in fact, resume the download.

Tested by injecting a fault in a locally-compiled XCache so it'd always error out at 10MB.  I then set the cache by hand to the local XCache, then turned on an `http_proxy` (this way, it'd be probed once with the proxy and once without).  I see the two requests come in, one via squid and one via grab directly ... but no byte ranges on the second.
